### PR TITLE
Stop rendering all DR's after refresh

### PR DESCRIPTION
**Pull request by dp6ai**
- use policies to limit what is added to the dashboard afte rthe refresh

### CODE REVIEW

File: spec/features/dashboard_spec.rb [hunk 1]
```diff
@@ -217,6 +219,28 @@
...
     end
+
+    scenario 'after the refresh i can still see ONLY the correct data', js: true do
+      visit defence_requests_path
+
+      within ".accepted_defence_request" do
+        expect(page).to have_content(solicitor_dr.detainee_name)
+        expect(page).to_not have_content(solicitor_dr_not_accepted.detainee_name)
+      end
+      expect(page).to_not have_selector(".created_defence_request")
```
Comment (pixeltrix) [inline spec/features/dashboard_spec.rb]: Really don't like setting expectations on styles like this - makes tests very brittle to design changes, but I guess it's okay for the alpha

File: spec/features/dashboard_spec.rb [hunk 2]
```diff
@@ -217,6 +219,28 @@
...
+      expect(page).to_not have_selector(".open_defence_request")
+
+      sleep(4)
```
Comment (pixeltrix) [inline spec/features/dashboard_spec.rb]: What's the reason to sleep here?

File: spec/features/dashboard_spec.rb (hunk 1 above)
Comment (dp6ai) [inline spec/features/dashboard_spec.rb]: Agree. We’re doing this a lot and its not ideal. One of the tasks next week is to refactor/clean all this up.




Patrick Davenport		
Unboxed
Consulting
E:	patrick.davenport@unboxedconsulting.com
M:	+44 7964 292 110
T:	+44 20 7183 4264
F:	+44 20 7183 4251
17 Blossom Street
London, E1 6PL
United Kingdom
www.unboxedconsulting.com

On 24 February 2015 at 10:33:23, Andrew White (notifications@github.com) wrote:

In spec/features/dashboard_spec.rb:

> @@ -217,6 +219,28 @@
>          expect(page).to_not have_content(other_solicitor_dr.solicitor_name)
>        end
>      end
> +
> +    scenario 'after the refresh i can still see ONLY the correct data', js: true do
> +      visit defence_requests_path
> +
> +      within ".accepted_defence_request" do
> +        expect(page).to have_content(solicitor_dr.detainee_name)
> +        expect(page).to_not have_content(solicitor_dr_not_accepted.detainee_name)
> +      end
> +      expect(page).to_not have_selector(".created_defence_request")
Really don't like setting expectations on styles like this - makes tests very brittle to design changes, but I guess it's okay for the alpha

File: spec/features/dashboard_spec.rb (hunk 2 above)
Comment (dp6ai) [inline spec/features/dashboard_spec.rb]: We’ve had an ongoing issue with javascript driven tests failing intermittently on CI this fixed it but again theres a task to sort it next week. 



Patrick Davenport		
Unboxed
Consulting
E:	patrick.davenport@unboxedconsulting.com
M:	+44 7964 292 110
T:	+44 20 7183 4264
F:	+44 20 7183 4251
17 Blossom Street
London, E1 6PL
United Kingdom
www.unboxedconsulting.com

On 24 February 2015 at 10:35:00, Andrew White (notifications@github.com) wrote:

In spec/features/dashboard_spec.rb:

> @@ -217,6 +219,28 @@
>          expect(page).to_not have_content(other_solicitor_dr.solicitor_name)
>        end
>      end
> +
> +    scenario 'after the refresh i can still see ONLY the correct data', js: true do
> +      visit defence_requests_path
> +
> +      within ".accepted_defence_request" do
> +        expect(page).to have_content(solicitor_dr.detainee_name)
> +        expect(page).to_not have_content(solicitor_dr_not_accepted.detainee_name)
> +      end
> +      expect(page).to_not have_selector(".created_defence_request")
> +      expect(page).to_not have_selector(".open_defence_request")
> +
> +      sleep(4)
What's the reason to sleep here?

**Pull request merged**